src/onyx/state/bookkeeper.clj

```diff
@@ -4,71 +4,80 @@
+        cookie-path (format "%s/cookies/%s"
+                            (.getZkLedgersRootPath server-conf)
+                            (Bookie/getBookieAddress server-conf))]
+    (info "Deleting Bookie cookie" cookie-path)
+    (zk/delete zk-conn cookie-path)
+    (info "Deleting Bookie dirs" journal-dir ";" ledger-dir)
+    (cleanup-dir journal-dir)
+    (cleanup-dir ledger-dir)))
+
 (defrecord BookieComponent [env-config port log]
   component/Lifecycle
   (start [component]
-    (let [onyx-id (:onyx/tenancy-id env-config)
-          ledgers-root-path (ozk/ledgers-path onyx-id)
-          ledgers-available-path (ozk/ledgers-available-path onyx-id)
-          _ (zk/create (:conn log) ledgers-root-path :persistent? true)
-          _ (zk/create (:conn log) ledgers-available-path :persistent? true)
-          base-journal-dir (arg-or-default :onyx.bookkeeper/base-journal-dir env-config)
-          base-ledger-dir (arg-or-default :onyx.bookkeeper/base-ledger-dir env-config)
-          ;; allow loopback? only if running a local quorum
-          allow-loopback? (boolean (arg-or-default :onyx.bookkeeper/local-quorum? env-config))
-          disk-usage-threshold (arg-or-default :onyx.bookkeeper/disk-usage-threshold env-config)
-          disk-usage-warn-threshold (arg-or-default :onyx.bookkeeper/disk-usage-warn-threshold env-config)
-          server-id (str onyx-id "_" port)
-          journal-dir (str base-journal-dir "/" server-id)
-          ledger-dir (str base-ledger-dir "/" server-id)
-          server-conf (doto (ServerConfiguration.)
-                        (.setZkServers (:zookeeper/address env-config))
-                        (.setZkLedgersRootPath ledgers-root-path)
-                        (.setBookiePort port)
-                        (.setJournalDirName journal-dir)
-                        (.setLedgerDirNames (into-array String [ledger-dir]))
-                        (.setAllowLoopback allow-loopback?)
-                        (.setDiskUsageThreshold disk-usage-threshold)
-                        (.setDiskUsageWarnThreshold disk-usage-warn-threshold))
-          server (try (BookieServer. server-conf)
-                      (catch Exception e
-                        (if (instance? KeeperException$NodeExistsException (.getCause e))
-                          (let [cookie-path (format "%s/cookies/%s"
-                                                    ledgers-root-path
-                                                    (Bookie/getBookieAddress server-conf))]
-                            (info "Deleting existing Bookie cookie" cookie-path)
-                            (zk/delete (:conn log) cookie-path)
-                            (BookieServer. server-conf))
-                          (throw e))))]
-      (info "Starting BookKeeper server on port" port)
-      (.start ^BookieServer server)
-      (when (:onyx.bookkeeper/delete-server-data? env-config)
-        (.addShutdownHook (Runtime/getRuntime)
```

Comment:
I think we should keep the shutdown hook. Sometimes components aren't shutdown properly, but we still want to format it.